sklearn/linear_model/tests/test_logistic.py

```diff
@@ -90,6 +91,18 @@ def test_predict_3_classes():
     check_predictions(LogisticRegression(C=10), X_sp, Y2)
 
 
+def test_lr_liblinear_warning():
+    n_samples, n_features = iris.data.shape
+    target = iris.target_names[iris.target]
+
+    for LR in [LogisticRegression(solver='liblinear', n_jobs=2),
+               LogisticRegressionCV(solver='liblinear', n_jobs=2)]:
```

Comment:
hmm... I myself didn't change `test_logistic.py`. Based on the snippets shown in your comment, seems it's a case added to test the display of warning. Maybe @glemaitre have some comments on that. 

Based on the [fit](https://github.com/zyxue/scikit-learn/blob/497b59a4d4a7763ae00e046a163add6bbb57b5bc/sklearn/linear_model/logistic.py#L1220) function, you could see that the method returns early when detecting the solver is `liblinear` without using `n_jobs`. My experience is also that `n_jobs` has no effect on multiprocessing (I didn't test for the other solvers), which is what motivation behind this pull request so that others won't need to wonder as I did.